File: backjoon/back-11724.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath).toString().trim().split("\r\n");
let [N, M] = input[0].split(" ").map(Number);

let graph = [];
for (let i = 0; i <= N; i++) {
  graph[i] = [];
}
let visited = new Array(N + 1).fill(false);
input.shift();
input.map((el) => {
  let [A, B] = el.split(" ").map(Number);
  graph[A].push(B);
  graph[B].push(A);
});

function dfs(v) {
  visited[v] = true;
  for (const el of graph[v]) {
    if (!visited[el]) {
      dfs(el);
    }
  }
}

let ans = 0;
for (let i = 1; i <= N; i++) {
  if (!visited[i]) {
    dfs(i);
    ans++;
  }
}

console.log(ans);

File: back-python/back-11724.py
import sys
sys.setrecursionlimit(10000)
sys.stdin = open("input_py.txt", "r")

def dfs(v):
  visited[v] = True  
  for i in graph[v]:
    if not visited[i]:
      dfs(i)

N, M  = map(int,sys.stdin.readline().split())
graph = [[] for _ in range(N+1)]
visited = [False] * (N+1)
cnt = 0

for _ in range(M):
  u, v  = map(int, sys.stdin.readline().split())
  graph[u].append(v)
  graph[v].append(u)

for i in range(1, N+1):
  if not visited[i]:
    dfs(i)    
    cnt +=1

print(cnt)